### PR TITLE
chore(CLI): start working on CLI

### PR DESCRIPTION
**Pull request by pedrooot**
### Description

In this PR new CLI it's added, I'll be improving this but at the momment It just list services and shows the banner.

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cli/cli.py
Comment (jfagoagas) [inline cli/cli.py:1]: Add an `__init__.py` file in this folder please.

File: cli/cli.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    print_services(list_services(provider))
+
+
+@app.command("banner", help="Prints the banner of the tool.")
+def banner(show: bool = True):
+    total = 0
+    for value in track(range(100), description="Processing..."):
+        time.sleep(0.01)
+        total += 1
+    if show:
```
Comment (jfagoagas) [inline cli/cli.py]: Do we need this?

File: pyproject.toml [hunk 2]
```diff
@@ -70,6 +70,12 @@ slack-sdk = "3.27.2"
 tabulate = "0.9.0"
 tzlocal = "5.2"
 
+[tool.poetry.group.cli]
+optional = true
+
+[tool.poetry.group.cli.dependencies]
+typer = "^0.12.3"
```
Comment (jfagoagas) [inline pyproject.toml]:
```suggestion
typer = "0.12.3"
```

**Review by jfagoagas — CHANGES_REQUESTED** [review]

File: cli/cli.py (hunk 1 above)
Comment (pedrooot) [reply cli/cli.py]: Not really, it's to show the status bar

File: pyproject.toml (hunk 2 above)
Comment (pedrooot) [reply pyproject.toml]: Done

File: cli/cli.py (hunk 1 above)
Comment (pedrooot) [reply cli/cli.py]: Update: deleted

File: cli/cli.py
Comment (pedrooot) [reply cli/cli.py:1]: Done

File: cli/cli.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+)
+def list_services_gcp():
+    print_services(list_services("gcp"))
+
+
+@kubernetes.command(
+    "list-services", help="List the Kubernetes services that are supported by Prowler."
+)
+def list_services_kubernetes():
+    print_services(list_services("kubernetes"))
```
Comment (jfagoagas) [inline cli/cli.py]: Is there any way to make this a generic command for all the providers in typer?

**Review by jfagoagas — CHANGES_REQUESTED** [review]

File: cli/cli.py (hunk 3 above)
Comment (pedrooot) [reply cli/cli.py]: Done

File: cli/cli.py [hunk 4]
```diff
@@ -1,45 +1,52 @@
...
 
-@azure.command(
-    "list-services", help="List the Azure services that are supported by Prowler."
-)
-def list_services_azure():
-    print_services(list_services("azure"))
+def create_list_commands(provider_typer: typer.Typer, provider_name: str):
+    @provider_typer.command(
+        "list-services",
+        help=f"List the {provider_name} services that are supported by Prowler.",
```
Comment (jfagoagas) [inline cli/cli.py:35]: Can you get the `provider_name` from `typer.Typer`?

Comment (pedrooot) [reply cli/cli.py:35]: Yes, done

**Review by jfagoagas — APPROVED** [review]
🔝

**Pull request merged**